controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -277,33 +262,31 @@ func (w *Workload) UpdateKubeletConfigMap(ctx context.Context, version semver.Ve
 	// NOTE: It is considered safe to update the kubelet-config-1.21 ConfigMap
 	// because only new nodes using v1.21 images will pick up the change during
 	// kubeadm join.
-	if version.GE(minVerKubeletSystemdDriver) {
-		data, ok := cm.Data[kubeletConfigKey]
-		if !ok {
-			return errors.Errorf("unable to find %q key in %s", kubeletConfigKey, cm.Name)
-		}
-		kubeletConfig, err := yamlToUnstructured([]byte(data))
-		if err != nil {
-			return errors.Wrapf(err, "unable to decode kubelet ConfigMap's %q content to Unstructured object", kubeletConfigKey)
-		}
-		cgroupDriver, _, err := unstructured.NestedString(kubeletConfig.UnstructuredContent(), cgroupDriverKey)
-		if err != nil {
-			return errors.Wrapf(err, "unable to extract %q from Kubelet ConfigMap's %q", cgroupDriverKey, cm.Name)
-		}
+	data, ok := cm.Data[kubeletConfigKey]
```

Comment:
wondering if the cgroup driver override can be removed at this point.
existing clusters were already mutated by this code, while for new clusters kubeadm does the defaulting.

it might be safer to do this once:
https://github.com/kubernetes/enhancements/blob/7e1c131dc50fa84caa5e99da2b8bf67db32054ab/keps/sig-node/4033-group-driver-detection-over-cri/kep.yaml
goes GA.

however CR need to adapt it first too.

Reply:
Added it to https://github.com/kubernetes-sigs/cluster-api/issues/8190 (to consider this later)